lib/internal/vm/module.js

```diff
@@ -394,7 +395,7 @@ class SyntheticModule extends Module {
     const { context, identifier } = options;
 
     super({
-      syntheticExportNames: exportNames,
+      syntheticExportNames: [...new Set(exportNames)],
```

Comment:
Could this throw if there are duplicate names rather than silently ignore? I think it's a user error and should be surfaced

Reply:
Something like

```js
const syntheticExportNames = [];

for (const name of exportNames) {
  if (syntheticExportNames.includes(name)) {
    throw new Error(`Export name "${name}" was provided multiple times`);
  }
  syntheticExportNames.push(name);
}
```